eng/packages/http-client-csharp/generator/Azure.Generator/src/Providers/MgmtLongRunningOperationProvider.cs

```diff
@@ -0,0 +1,226 @@
+using Azure.Core.Pipeline;
+using Azure.Generator.Utilities;
+using Azure.ResourceManager;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Providers;
+using Microsoft.Generator.CSharp.Statements;
+using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static Microsoft.Generator.CSharp.Snippets.Snippet;
+
+namespace Azure.Generator.Providers
+{
+    internal class MgmtLongRunningOperationProvider : TypeProvider
+    {
+        private class Template<T> { }
+        private readonly CSharpType _t = typeof(Template<>).GetGenericArguments()[0];
+
+        private bool _isGeneric;
+        private FieldProvider _operationField;
+        private FieldProvider _rehydrationTokenField;
+        private FieldProvider _nextLinkOperationField;
+        private FieldProvider _operationIdField;
+
+        public MgmtLongRunningOperationProvider(bool isGeneric)
+        {
+            _isGeneric = isGeneric;
+            _operationField = new FieldProvider(FieldModifiers.Private | FieldModifiers.ReadOnly, isGeneric ? new CSharpType(typeof(OperationInternal), _t) : typeof(OperationInternal), "_operation", this);
+            _rehydrationTokenField = new FieldProvider(FieldModifiers.Private | FieldModifiers.ReadOnly, new CSharpType(typeof(RehydrationToken), true), "_completeRehydrationToken", this);
+            _nextLinkOperationField = new FieldProvider(FieldModifiers.Private | FieldModifiers.ReadOnly, new CSharpType(typeof(NextLinkOperationImplementation), true), "_nextLinkOperation", this);
+            _operationIdField = new FieldProvider(FieldModifiers.Private | FieldModifiers.ReadOnly, typeof(string), "_operationId", this);
+        }
+
+        private readonly string _serviceName = AzureClientPlugin.Instance.InputLibrary.InputNamespace.Name.Split('.').Last();
+        protected override string BuildName() => /*_isGeneric ? $"{_serviceName}ArmOperation<T>" : */$"{_serviceName}ArmOperation";
+
+        protected override CSharpType[] GetTypeArguments() => _isGeneric ? new CSharpType[] { _t } : base.GetTypeArguments();
+
+        protected override string BuildRelativeFilePath()
+        {
+            return Path.Combine("src", "Generated", "LongRunningOperation", GetFileName());
+
+            string GetFileName() => _isGeneric ? $"{_serviceName}ArmOperationOfT.cs" : $"{Name}.cs";
+        }
+
+        protected override CSharpType[] BuildImplements() => [_isGeneric ? new CSharpType(typeof(ArmOperation), _t) : typeof(ArmOperation)];
+
+        protected override FieldProvider[] BuildFields()
+            => [_operationField, _rehydrationTokenField, _nextLinkOperationField, _operationIdField];
+
+        protected override MethodProvider[] BuildMethods()
+        {
+            var getRehydrationTokenMethod = new MethodProvider(
```

Comment:
nit: can we add an empty line between each different method providers?

Reply:
sure